Removing generate config file in favor of GitHub secrets

jasonnrosa opened the pull request:
**Types of change**

- [ ] Bug fix (change which fixes an issue)
- [X] Enhancement (change which improves upon an existing feature)
- [ ] New feature (change which adds new functionality)

**Description**

- Updating the upload command in `deploy.sh` as `npx` is no longer used. 
- Removing generate config file per request here (#199). We have HubSpot portal ID and HubSpot API key set in [.github/workflows/deploy.yml](https://github.com/HubSpot/cms-theme-boilerplate/blob/master/.github/workflows/deploy.yml#L19) which should cover what was required [here](https://github.com/HubSpot/hubspot-cms-tools/pull/118) for the API key requirements.

**Questions**

- Should I remove `bin/generate-config.js` under `run` in [`deploy.yml`](https://github.com/HubSpot/cms-theme-boilerplate/blob/master/.github/workflows/deploy.yml#L23)?
- Is there a need for the `js-yaml` dev dependency in [`package.json`](https://github.com/HubSpot/cms-theme-boilerplate/blob/master/package.json#L14) if we remove the [generate config file](https://github.com/HubSpot/cms-theme-boilerplate/blob/master/bin/generate-config.js#L5)? 

**Relevant links**

Fixes #199 

**Checklist**

- [X] I have read the [CONTRIBUTING](https://github.com/HubSpot/cms-theme-boilerplate/blob/master/CONTRIBUTING.md) document.
- [X] My code follows the [style guide requirements](https://github.com/HubSpot/cms-theme-boilerplate/blob/master/STYLEGUIDE.md).
- [X] I have thoroughly tested my change.

**People to notify**
CC: @TheWebTech @ajlaporte @ckconant @TanyaScales

gcorne commented:
This should wait on https://github.com/HubSpot/hubspot-cms-tools/pull/317 and an official release. cc @miketalley

gcorne commented on bin/deploy.sh:
```diff
@@ -1,2 +1,2 @@
 #!/bin/bash
-npx hs upload src cms-theme-boilerplate
```
FWIW, I am pretty sure that `npx` is needed because the CLI is not installed globally as part of the GitHub action.

gcorne commented:
As part of this, I think we also should switch over to using a personal access key instead of API key. It is less risky since we can limit the scope of what the access key can do.

anthmatic commented:
@jasonnrosa I'm working on getting this https://github.com/HubSpot/hubspot-cms-deploy-action finished up today which should make this simpler to implement on your end

jasonnrosa commented:
To do: switch to using new CMS deploy action referenced here (https://github.com/marketplace/actions/hubspot-cms-deploy).

jasonnrosa commented:
Hey @gcorne @anthmatic @miketalley just updated boilerplate with this PR to use (https://github.com/marketplace/actions/hubspot-cms-deploy). I removed the `generate-config.js` and `deploy.sh` that were previously in the `bin` folder. I added two new GitHub secrets (personal access key and a second portal ID secret that follows the formatting mentioned in the documentation above). Then I updated the existing `deploy.sh` to match the documentation as well. Once this is all set I'll remove the old Portal ID secret and the old API key secret.
